generate.sh

```diff
@@ -108,8 +183,9 @@ echo "func Test${1}(t *testing.T) {
 	} {
 "
 
-grep -oP "[\t ]+[A-Z0-9_]+[\t ]+=[\t ]+[\-0-9xa-f <]+" | \
-	awk '{val=$1;$1=$2="";printf("\t\t{iproto.%s, \"%s\",%s},\n", val, val, $0)}'
+grep -oP "[\t ]*[A-Z0-9_]+[\t ]+=[\t ]+[A-Z_]*[\-0-9xa-f <\+]+" | \
+	awk '{val=$1;$1=$2="";gsub(/[A-Z][A-Z0-9_]+/,"int(iproto.&)",$0);
```

Comment:
Why not using only awk match? e.g.:
awk '/match/ {code}'

Reply:
I don't think it can be done only by using awk. We need to leave only matched string (but not the whole line it is in). In the current code `grep -o` is doing that, and its output we pass to the awk.